File: TheaterSuggestions/CSharp/TheaterSuggestions.Tests/Seat.cs
namespace SeatsSuggestions.Tests
{
    public class Seat
    {
        public Seat(string rowName, uint number, PricingCategory pricingCategory, SeatAvailability seatAvailability)
        {
            RowName = rowName;
            Number = number;
            PricingCategory = pricingCategory;
            SeatAvailability = seatAvailability;
        }

        public string RowName { get; }
        public uint Number { get; }
        public PricingCategory PricingCategory { get; }
        private SeatAvailability SeatAvailability { get; set; }

        public bool IsAvailable()
        {
            return SeatAvailability == SeatAvailability.Available;
        }

        public override string ToString()
        {
            return $"{RowName}{Number}";
        }

        public void UpdateCategory(SeatAvailability seatAvailability)
        {
            SeatAvailability = seatAvailability;
        }

        public bool MatchCategory(PricingCategory pricingCategory)
        {
            if (pricingCategory == PricingCategory.Mixed)
            {
                return true;
            }

            return PricingCategory == pricingCategory;
        }

        public void MarkAsAlreadySuggested()
        {
            if (SeatAvailability == SeatAvailability.Available)
            {
                SeatAvailability = SeatAvailability.Suggested;
            }
        }
    }
}

File: TheaterSuggestions/Java/SeatsSuggestionsTests/src/test/java/com/baasie/SeatsSuggestionsTests/Seat.java
package com.baasie.SeatsSuggestionsTests;

public class Seat {

    private String rowName;
    private int number;
    private PricingCategory pricingCategory;
    private SeatAvailability seatAvailability;

    public Seat(String rowName, int number, PricingCategory pricingCategory, SeatAvailability seatAvailability) {
        this.rowName = rowName;
        this.number = number;
        this.pricingCategory = pricingCategory;
        this.seatAvailability = seatAvailability;
    }

    public boolean isAvailable() {
        return seatAvailability == SeatAvailability.Available;
    }

    public boolean matchCategory(PricingCategory pricingCategory) {
        if (pricingCategory == PricingCategory.Mixed) {
            return true;
        }

        return this.pricingCategory == pricingCategory;
    }

    public void markAsAlreadySuggested() {
        if (seatAvailability == SeatAvailability.Available) {
            seatAvailability = SeatAvailability.Suggested;
        }
    }

    @Override
    public String toString() {
        return rowName + number;
    }
}
